storage/zfs/zfs-tools/pkg.yaml

```diff
@@ -0,0 +1,61 @@
+name: zfs-tools
+variant: scratch
+shell: /toolchain/bin/bash
+dependencies:
+  - stage: base
+steps:
+  - sources:
+      - url: https://github.com/openzfs/zfs/releases/download/zfs-{{ .ZFS_DRIVER_VERSION }}/zfs-{{ .ZFS_DRIVER_VERSION }}.tar.gz
+        destination: zfs.tar.gz
+        sha256: 64daa26aed3e12c931f6f4413d7527c4ebdb8da35416b356152b5f9fdd4c6e6d
+        sha512: f48493a21883e441cda705fb085353bed033f1620a1d0f93069c345c76cf2c0759a2e6f7a80c47c9398e9878abfe1d90d931fe5ceaf2588770a71491a434631e
+      - url: https://downloads.sourceforge.net/project/libtirpc/libtirpc/1.3.3/libtirpc-1.3.3.tar.bz2
+        destination: libtirpc.tar.bz2
+        sha256: 6474e98851d9f6f33871957ddee9714fdcd9d8a5ee9abb5a98d63ea2e60e12f3
+        sha512: df0781a74ff9ded2d3c4f5eb7e05496b9f58eac8060c02c68331dc14c4a00304dcd19f46836f5756fe0d9d27095fd463d42dd696fcdff891516711b7d63deabe
+    env:
+      ARCH: {{ if eq .ARCH "aarch64"}}arm64{{ else if eq .ARCH "x86_64" }}x86_64{{ else }}unsupported{{ end }}
+    prepare:
+      - |
+        cp /pkg/files/sys-queue.h /usr/include/sys/queue.h
+        mkdir libtirpc
+        tar -jxf libtirpc.tar.bz2 --strip-components=1 -C libtirpc
+        tar -zxf zfs.tar.gz --strip-components=1
+    build:
+      - |
+        cd libtirpc
+        ./configure \
+          --prefix=/usr/local \
+          --disable-shared \
+          --disable-gssapi
+        make -j $(nproc) install
+        cd ..
+
+        ./configure \
+          LDFLAGS="--static" \
+          --prefix=/usr/local \
+          --sysconfdir=/etc \
+          --with-udevdir=/usr/etc/udev \
+          --with-linux=/src \
+          --with-config=user \
+          --disable-shared
+
+        sed -i 's/-all-static/--static/g' libtool
+
+        make -j $(nproc)
+    install:
+      - |
+        make install-exec DESTDIR=/rootfs/tmpdir
+        make install-data DESTDIR=/rootfs/tmpdir
+        mkdir -p /rootfs/usr/local/{sbin,bin}
+        mkdir -p /rootfs/usr/etc/udev/rules.d/
+        cp -R /rootfs/tmpdir/usr/etc/udev/rules.d/* /rootfs/usr/etc/udev/rules.d
+        sed -i s,/usr/etc/udev/,/usr/local/sbin/,g /rootfs/usr/etc/udev/rules.d/*
+        mv /rootfs/tmpdir/usr/etc/udev/zvol_id /rootfs/usr/local/sbin/zvol_id
```

Comment:
Thanks for reviewing the patch!
I'm a bit puzzled when you say `these udev rules are not going to work since there is no shell in talos` and removed those.
We added those for a reason ;-)
Upon zvol creation in the zpool, those udev rules are creating symlink in /dev (and need the using the zvol_id binary for that matter, that's also why we put it there): It's required for openebs-zfs-localpv to work! (when the zvol is created, the CSI will try to format & mount the zpool block device using the created symlink.

Reply:
This was a wrong analysis, `zvol_id` is a binary, added the rules back based on slack thread.